### DISCUSSION TITLE
System.Text.Rune

### DISCUSSION THREAD

**Discussion by brainoffline** [tui-cs/Terminal.Gui#1493] [Ideas]
A Rune is now a native struct from .Net Core 3.0 +.
This could mean removing the dependency on the NStack package.
But... System.Text.Rune is not available in .Net Framework or .Net Standard.

What is the current feeling about a breaking change and leaving .net Framework and standard behind?

Or, some kind of port of the NStack ustring to expose the same interfaces as .Net Core 3.0+ string.

**Comment by tznind** [comment]
> What is the current feeling about a breaking change and leaving .net Framework and standard behind?

I don't think we should be dropping support for older frameworks without a really good reason.

**Reply by brainoffline** [reply]
Yeah. We struggle with the same question with every mobile release.  We officially support <T> -2 and that does help.  But for open source, that is a different question.

**Comment by tig** [comment]
If it ain't broke, don't fix it.

I can't recall a single issue with nstack's Rune support in the past 2 years.
